lib/services/android-project-service.ts

```diff
@@ -396,18 +399,24 @@ export class AndroidProjectService extends projectServiceBaseLib.PlatformProject
 
 			let buildOptions = this.getBuildOptions();
 
-			buildOptions.unshift("clean");
-
 			let projectRoot = this.platformData.projectRoot;
-			let gradleBin = this.useGradleWrapper(projectRoot) ? path.join(projectRoot, "gradlew") : "gradle";
-			if (this.$hostInfo.isWindows) {
-				gradleBin += ".bat";
-			}
-			this.spawn(gradleBin, buildOptions, { stdio: "inherit", cwd: this.platformData.projectRoot }).wait();
+
+			this.cleanProject(projectRoot, buildOptions);
 		}
 		return Future.fromResult();
 	}
 
+	private cleanProject(projectRoot: string, options: string[]): void {
```

Comment:
the method has async operation (`.wait()` ), so it must return `IFuture<void>` instead of `void`.